backend/src/worker.rs

```diff
@@ -530,138 +551,514 @@ async fn handle_nondep_job(
-                        .stderr(Stdio::piped())
-                        .spawn()?
-                } else {
-                    Command::new(python_path)
-                        .current_dir(job_dir)
-                        .args(vec![
-                            "-m",
-                            "pip",
-                            "install",
-                            "--no-color",
-                            "--isolated",
-                            "--no-warn-conflicts",
-                            "--disable-pip-version-check",
-                            "-t",
-                            "./dependencies",
-                            "-r",
-                            "./requirements.txt",
-                        ])
-                        .stdout(Stdio::piped())
-                        .stderr(Stdio::piped())
-                        .spawn()?
-                };
+        install_go_dependencies(
+            &job.id,
+            inner_content,
+            logs,
+            job_dir,
+            status,
+            db,
+            last_line,
+            timeout,
+            go_path,
+            true,
+        )
+        .await?;
+    }
 
-                logs.push_str("\n--- PIP DEPENDENCIES INSTALL ---\n");
-                *status = handle_child(job, db, logs, last_line, timeout, child).await;
-                tracing::info!(
-                    worker_name = %worker_name,
-                    job_id = %job.id,
-                    workspace_id = %job.workspace_id,
-                    is_ok = status.is_ok(),
-                    "finished setting up python dependencies {}",
-                    job.id
-                );
-            }
-            if requirements.len() == 0 || status.is_ok() {
-                logs.push_str("\n\n--- PYTHON CODE EXECUTION ---\n");
+    logs.push_str("\n\n--- GO CODE EXECUTION ---\n");
+    set_logs(logs, job.id, db).await;
+
+    let sig = crate::parser_go::parse_go_sig(&inner_content)?;
+    let token = create_token_for_owner(
+        &db,
+        &job.workspace_id,
+        &job.permissioned_as,
+        "ephemeral-script",
+        timeout * 2,
+        &job.created_by,
+    )
+    .await?;
+    let args = if let Some(args) = &job.args {
+        Some(
+            transform_json_value(&token, &job.workspace_id, &base_internal_url, args.clone())
+                .await?,
+        )
+    } else {
+        None
+    };
+    let ser_args = serde_json::to_string(&args).map_err(|e| Error::ExecutionErr(e.to_string()))?;
+    write_file(job_dir, "args.json", &ser_args).await?;
+    let spread = sig
+        .args
+        .into_iter()
+        .map(|x| {
+            format!(
+                "json_arg[\"{}\"].({})",
+                x.name,
+                x.otyp.unwrap_or_else(|| "interface{}".to_string())
+            )
+        })
+        .join(", ");
+
+    let inner_content = inner_content.replace("func main(", "func inner_main(");
```

Comment:
This doesn't seem particularly robust as it may match other occurrences of that string that it shouldn't. Also it won't match "func     main(" (which doesn't seem to get parsed correctly right now anyway).

Reply:
>  Also it won't match "func main(" (which doesn't seem to get parsed correctly right now anyway).

Not sure to understand what you mean

> This doesn't seem particularly robust as it may match other occurrences of that string that it shouldn't.

That's true, we have a full parser so I could use that one but it could be rather tedious to do it at that level. Suggestions ?

Reply:
> Not sure to understand what you mean

`"func           main("` where there is whitespace between func and main. (github collapsed whitespace even though it looked fine in preview)